src/weakptr.c

```diff
@@ -823,8 +819,7 @@ void CleanObjWPObjCopy (
     Obj                 elm;            /* subobject                       */
 
     /* remove the forwarding pointer                                       */
-    ADDR_OBJ(obj)[0] = ELM_PLIST( CONST_ADDR_OBJ(obj)[0], 1 );
-    CHANGED_BAG(obj);
```

Comment:
Note that the `CHANGED_BAG` here is dropped intentionally: the value restored to `obj` is a raw integer (the "stored length" of the wp obj, and never a bag ref.